Looms.PoS/Looms.PoS.Persistance/Repositories/OrdersRepository.cs

```diff
@@ -0,0 +1,134 @@
+using Looms.PoS.Domain.Daos;
+using Looms.PoS.Domain.Exceptions;
+using Looms.PoS.Domain.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Looms.PoS.Domain.Filters.Order;
+using Looms.PoS.Domain.Enums;
+
+namespace Looms.PoS.Persistance.Repositories;
+
+public class OrdersRepository : LoomsException, IOrdersRepository
+{
+    private readonly AppDbContext _context;
+
+    public OrdersRepository(AppDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<OrderDao> CreateAsync(OrderDao order)
+    {
+        var orderDao = await _context.Orders.AddAsync(order);
+        await _context.SaveChangesAsync();
+        return orderDao.Entity;
+    }
+
+    public async Task RemoveAsync(Guid id)
+    {
+        var order = await _context.Orders.FindAsync(id);
+        _context.Orders.Remove(order!);
+    }
+
+    public async Task<IEnumerable<OrderDao>> GetAllAsync()
+    {
+        return await _context.Orders
+            .Where(x => !x.IsDeleted)
+            .Include(x => x.Discount)
+            .Include(x => x.Payments)
+            .Include(x => x.Refunds)
+            .Include(x => x.Business)
+            .Include(x => x.OrderItems.Where(oi => !oi.IsDeleted))
+                .ThenInclude(oi => oi.ProductVariation)
+            .Include(x => x.OrderItems.Where(oi => !oi.IsDeleted))
+                .ThenInclude(oi => oi.Discount)
+            .Include(x => x.OrderItems.Where(oi => !oi.IsDeleted))
+                .ThenInclude(oi => oi.Product)
+                    .ThenInclude(p => p.Tax)
+            .ToListAsync();
+    }
+
+    public async Task<IEnumerable<OrderDao>> GetAllAsync(GetAllOrdersFilter filter)
+    {
+        var query = _context.Orders
+            .Where(x => !x.IsDeleted);
+
+        if(filter.Status is not null)
+        {
+            query = query.Where(x => x.Status == Enum.Parse<OrderStatus>(filter.Status));
+        }
+
+        if(filter.UserId is not null)
+        {
+            query = query.Where(x => x.UserId == Guid.Parse(filter.UserId));
+        }
```

Comment:
Status and userId are never null

Reply:
Because these are query parameters that means if they are used, they have to have a value, right?

Reply:
I don't know how .net handles query parameters, but I would assume that if there is no query parameter, property should be null. However, these checks check nothing as by definition Status and UserId are never null. IDE should highlight it.

Reply:
<img width="721" alt="image" src="https://github.com/user-attachments/assets/bd7161ad-1553-4056-8285-03222d1e9d42" />
<img width="364" alt="image" src="https://github.com/user-attachments/assets/a5c8c998-a558-4737-9b11-322118826a5f" />

Reply:
removed checks and now added just 
`query.Where(x => x.Status == Enum.Parse<OrderStatus>(filter.Status) && x.UserId == Guid.Parse(filter.UserId));`
but wont now either both query parameters are used or none?